cpp/include/Ice/InputStream.h

```diff
@@ -650,6 +651,26 @@ class ICE_API InputStream : public IceInternal::Buffer
         }
     }
 
+    /**
+     * Reads an optional proxy from the stream.
+     * @param tag The tag ID.
+     * @param v The proxy unmarshaled by this function. If nullopt, the proxy was not present in the stream or
+     * was set to nullopt (set to nullopt is supported for backward compatibility with Ice 3.7 and earlier releases).
+     */
+    template<typename T, std::enable_if_t<std::is_base_of<ObjectPrx, T>::value, bool> = true>
+    void read(std::int32_t tag, std::optional<T>& v)
+    {
+        if (readOptional(tag, OptionalFormat::FSize))
```

Comment:
I could not reuse the shared logic from the standard `read` above with a new StreamOptionalHelper specialization because Prx are unlikely every other types - they don't have a default ctor. So it's not possible to write:
```cpp
v.emplace(); // does not work when v is an std::optional<XxxPrx>
StreamOptionalHelper<T,
                                 StreamableTraits<T>::helper,
                                 StreamableTraits<T>::fixedLength>::read(this, *v);
```

The solution: provide a special read for tagged proxies.